db/migrate/20240715075900_add_admin_bot_protection_to_settings.rb

```diff
@@ -0,0 +1,10 @@
+class AddAdminBotProtectionToSettings < ActiveRecord::Migration[6.1]
+  def up
+    add_column :settings, :admin_bot_protection_level, :string
+    change_column_default :settings, :admin_bot_protection_level, "none"
```

Comment:
```suggestion
    add_column :settings, :admin_bot_protection_level, :string, default: "none"
```

Not sure if there was a specific reason to separate the two operations? If not, I think they can be merged into one.

Reply:
Yes, I did that at the beginning, but I received a warning from `strong_migrations` because it's and unsafe operation, only for Oracle.

```
=== Dangerous operation detected #strong_migrations ===

Adding a column with a non-null default causes the entire table to be rewritten.
Instead, add the column without a default value, then change the default.

class AddAdminBotProtectionToSettings < ActiveRecord::Migration[6.1]
  def up
    add_column :settings, :admin_bot_protection_level, :string
    change_column_default :settings, :admin_bot_protection_level, "none"
  end

  def down
    remove_column :settings, :admin_bot_protection_level
  end
end

Then backfill the existing rows in the Rails console or a separate migration with disable_ddl_transaction!.

class BackfillAddAdminBotProtectionToSettings < ActiveRecord::Migration[6.1]
  disable_ddl_transaction!

  def up
    Setting.unscoped.in_batches do |relation| 
      relation.update_all admin_bot_protection_level: "none"
      sleep(0.01)
    end
  end
end

Then add the NOT NULL constraint in separate migrations.

/home/jlledom/redhat/porta/db/migrate/20240711145208_add_admin_bot_protection_to_settings.rb:3:in `up'

```

So I update the migration to:
- Split into to operations
- Don't backfill
- Don't make the column `NOT NULL`
- Adapt the code to interpret `NULL` as `:none`

It's much simpler, less risky and works for all DB engines.

Reply:
OK, thanks for the explanation @jlledom